.github/actions/setup_build/action.yml

```diff
@@ -32,16 +33,31 @@ runs:
       if: ${{ inputs.os == 'ubuntu' }}
       shell: bash
       run: |
-        pip install lit clang-format==19.1.0 virtualenv
-        if [ "${{ inputs.arch }}" = "x86" ]; then sudo dpkg --add-architecture i386 ; fi
+        if [ "${{ inputs.cross_arch }}" = "x86" ]; then sudo dpkg --add-architecture i386 ; fi
         wget -qO - https://packages.lunarg.com/lunarg-signing-key-pub.asc | sudo apt-key add -
         if [ "${{ inputs.ubuntu_version }}" = "20.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-focal.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-focal.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "22.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-jammy.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-jammy.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "24.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-noble.list https://packages.lunarg.com/vulkan/lunarg-vulkan-noble.list; fi
         sudo apt-get update
+        pip install lit clang-format==19.1.0 virtualenv
         sudo apt-get install --yes doxygen
         sudo apt-get install --yes vulkan-sdk
-        if [ "${{ inputs.arch }}" = "x86" ]; then sudo apt-get install --yes gcc-multilib g++-multilib libc6-dev:i386 lib32tinfo-dev ; fi
+        if [ "${{ inputs.cross_arch }}" = "x86" ]; then \
+           sudo apt-get install --yes gcc-multilib g++-multilib libc6-dev:i386 lib32tinfo-dev; \
+        fi
+        if [ "${{ inputs.cross_arch }}" = "armhf" ]; then \
+           sudo apt-get install --yes gcc-11-arm-linux-gnueabihf g++-11-arm-linux-gnueabihf; \
+        fi
+        if [ "${{ inputs.cross_arch }}" = "aarch64" ]; then \
+           sudo apt-get install --yes gcc-11-aarch64-linux-gnu g++-11-aarch64-linux-gnu; \
+        fi
+        if [ "${{ inputs.cross_arch }}" = "riscv64" ]; then \
+           sudo apt-get install --yes gcc-11-riscv64-linux-gnu g++-11-riscv64-linux-gnu; \
+        fi
+        if [ "${{ inputs.cross_arch }}" = "armhf" -o "${{ inputs.cross_arch }}" = "aarch64" -o "${{ inputs.cross_arch }}" = "riscv64" ]; then \
```

Comment:
is it not easier to just say cross_arch != ""

Reply:
Done